### PR TITLE
feat(RDS): add rds pg database resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add rds pg database resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add rds pg database resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/rds/ TESTARGS='-run TestAccPgDatabase_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccPgDatabase_basic -timeout 360m -parallel 4
=== RUN   TestAccPgDatabase_basic
=== PAUSE TestAccPgDatabase_basic
=== CONT  TestAccPgDatabase_basic
--- PASS: TestAccPgDatabase_basic (462.47s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       462.507s
```

### CODE REVIEW

File: docs/resources/rds_pg_database.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  For details, see [documentation](https://www.postgresql.org/docs/16/infoschema-character-sets.html).
+  Defaults to **UTF8**.
+
+  Changing this parameter will create a new resource.
+
+* `lc_collate` - (Optional, String, ForceNew) Specifies the database collocation.
+  For details, see [documentation](https://support.huaweicloud.com/intl/en-us/bestpractice-rds/rds_pg_0017.html).
+  Defaults to **en_US.UTF-8**.
+
+-> **NOTE:** For different collation rules, the execution result of a statement may be different.
```
Comment (deer-hang) [inline docs/resources/rds_pg_database.md]: Please leave two spaces, if this document is a description of `lc_collate`, and the English description has grammatical errors.

File: docs/resources/rds_pg_database.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 30 minutes.
+* `update` - Default is 30 minutes.
+* `delete` - Default is 30 minutes.
+
+## Import
+
+The rds postgresql database can be imported using the `instance id` and `name` separated by slash, e.g.
```
Comment (deer-hang) [inline docs/resources/rds_pg_database.md]:
```suggestion
The RDS postgresql database can be imported using the `instance id` and `name` separated by a slash, e.g.
```

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_database_test.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+					resource.TestCheckResourceAttr(rName, "character_set", "UTF8"),
+					resource.TestCheckResourceAttr(rName, "lc_collate", "en_US.UTF-8"),
+					resource.TestCheckResourceAttr(rName, "description", "test_description_update"),
+				),
+			},
+			{
+				ResourceName:            rName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"template", "lc_ctype", "is_revoke_public_privilege"},
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_database_test.go:119]: Add description for import operation when ignoring fields.

Comment (houpeng80) [reply huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_database_test.go:119]: done

File: docs/resources/rds_pg_database.md (hunk 2 above)
Comment (houpeng80) [reply docs/resources/rds_pg_database.md]: done

File: docs/resources/rds_pg_database.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/rds_pg_database.md]: done

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_database.go [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Specifies whether to revoke the PUBLIC CREATE permission of the public schema.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
```
Comment (ShiChangkuo) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_database.go]: no computed, please

Comment (houpeng80) [reply huaweicloud/services/rds/resource_huaweicloud_rds_pg_database.go]: done

**Comment by houpeng80** [comment]
make testacc TEST=./huaweicloud/services/acceptance/rds/ TESTARGS='-run TestAccPgDatabase_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds/ -v -run TestAccPgDatabase_basic -timeout 360m -parallel 4
=== RUN   TestAccPgDatabase_basic
=== PAUSE TestAccPgDatabase_basic
=== CONT  TestAccPgDatabase_basic
--- PASS: TestAccPgDatabase_basic (521.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       521.362s

**Pull request merged**